src/test/java/net/minestom/server/coordinate/CoordinateTest.java

```diff
@@ -100,4 +100,18 @@ public void toSectionRelativeCoordinate() {
         assertEquals(0, ChunkUtils.toSectionRelativeCoordinate(32));
         assertEquals(1, ChunkUtils.toSectionRelativeCoordinate(33));
     }
+
+    @Test
+    public void testBlockIndex() {
```

Comment:
You need to test every possibility (ensuring that all coordinates give a different index, prevent potential overflow). There is otherwise no way to know if you broke something.

There is also no need to use `Pos`

Reply:
I have tried to get suitable test cases. I don't know if that are all but I also checked for Y overflow.

Y values between -2^23+1 to 2^23-1 convert correctly.
Everything above and below that results in a wrong conversion.

I don't know if I should catch this case and throw an exception, what do you think?
I think it is unlikely that someone wants and can use such high or low Y coordinates?

Reply:
Seems good to me